src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufSerializer.cs

```diff
@@ -0,0 +1,373 @@
+                {
+                    // TODO
+                    break;
+                }
+
+            case MetricType.ExponentialHistogram:
+                {
+                    // TODO
+                    break;
+                }
+        }
+    }
+
+    private void WriteIndividualMessageTagsAndLength(byte[] buffer, ref int currentPosition, MetricType metricType)
+    {
+        var instrumentIndex = this.previousIntrumentStartIndex;
+
+        var metricIndex = this.previousMetricStartIndex;
+
+        var scopeMetricsIndex = this.previousScopeMetricsStartIndex;
+
+        var resourceMetricIndex = this.resourceMetricStartIndex;
+
+        // Write instrument tag and length
+        ProtobufSerializerHelper.WriteTagAndLengthPrefix(buffer, ref instrumentIndex, currentPosition - instrumentIndex - LengthAndTagSize, FieldNumberConstants.GetMetricTypeFieldNumber(metricType), WireFormat.WireType.LengthDelimited);
+
+        // Write metric tag and length
+        ProtobufSerializerHelper.WriteTagAndLengthPrefix(buffer, ref metricIndex, currentPosition - metricIndex - LengthAndTagSize, FieldNumberConstants.ScopeMetrics_metrics, WireFormat.WireType.LengthDelimited);
+
+        // Write scope tag and length
+        ProtobufSerializerHelper.WriteTagAndLengthPrefix(buffer, ref scopeMetricsIndex, currentPosition - scopeMetricsIndex - LengthAndTagSize, FieldNumberConstants.ResourceMetrics_scope_metrics, WireFormat.WireType.LengthDelimited);
+
+        // Write resource metric tag and length
+        ProtobufSerializerHelper.WriteTagAndLengthPrefix(buffer, ref resourceMetricIndex, currentPosition - resourceMetricIndex - LengthAndTagSize, FieldNumberConstants.ResourceMetrics_resource, WireFormat.WireType.LengthDelimited);
+    }
+
+    private void SendMetricPoint(byte[] buffer, ref int currentPosition)
+    {
+        // TODO: Extend this for user_events.
+        this.MetricDataTransport.SendOtlpProtobufEvent(buffer, currentPosition);
+    }
+
+    internal static void WriteInstrumentDetails(byte[] buffer, ref int currentPosition, Metric metric)
+    {
+        // Write metric name
+        ProtobufSerializerHelper.WriteStringTag(buffer, ref currentPosition, metric.Name, FieldNumberConstants.Metric_name);
+
+        // Write metric description
+        if (metric.Description != null)
+        {
+            ProtobufSerializerHelper.WriteStringTag(buffer, ref currentPosition, metric.Description, FieldNumberConstants.Metric_description);
+        }
+
+        // Write metric unit
+        if (metric.Unit != null)
+        {
+            ProtobufSerializerHelper.WriteStringTag(buffer, ref currentPosition, metric.Unit, FieldNumberConstants.Metric_unit);
+        }
+    }
+
+    internal static void SerializeInstrumentationScope(byte[] buffer, ref int currentPosition, string name, IEnumerable<KeyValuePair<string, object>> meterTags)
+    {
+        int previousPosition = currentPosition;
+
+        currentPosition += LengthAndTagSize;
```

Comment:
Same issue as [this](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1596/files#r1529602334). Check if there are any meter tags first.

Reply:
This is slightly different. Instrumentation scope will always have the name field serialized even if there are no meter tags.

Reply:
> Instrumentation scope will always have the name field serialized even if there are no meter tags.

Why is that? The proto [file](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L69-L71) says:

"Semantically when InstrumentationScope isn't set, it is equivalent with an empty instrumentation scope name (unknown)."

It looks like it should be fine to not set the instrumentation scope.

Reply:
It's not an optional parameter in proto, we may need to set unknown when InstrumentationScope is not set

Reply:
InstrumentationScope has multiple fields. https://github.com/open-telemetry/opentelemetry-proto/blob/9d139c87b52669a3e2825b835dd828b57a455a55/opentelemetry/proto/common/v1/common.proto#L71-L80

So even if a scope(Meter) does not have attributes, we need to set the name/version fields.